src/lib/utils/isEqual.ts

```diff
@@ -0,0 +1,31 @@
+function isPrimitive(obj: any): boolean {
+  return obj !== Object(obj);
+}
+
+function isEqual(first: any, second: any): boolean {
+  if (first === second) {
+    return true;
+  }
+
+  if (isPrimitive(first) || isPrimitive(second)) {
+    return first === second;
+  }
+
+  if (Object.keys(first).length !== Object.keys(second).length) {
```

Comment:
This can throw in IE if first isn’t an object right?

Reply:
I tried from IE 9 to IE 11 and they all work as expected.

Reply:
TIL you can use Object.keys on Arrays!

Reply:
@Haroenv `undefined` & `null` fall inside the `isPrimitive` check.